Source/WTF/wtf/Vector.h

```diff
@@ -903,6 +904,19 @@ class Vector : private VectorBuffer<T, inlineCapacity, Malloc> {
     template<typename ResultVector, typename MapFunction>
     auto map(MapFunction&&) const -> std::enable_if_t<std::is_invocable_v<MapFunction, const T&>, ResultVector>;
 
+    template <typename BinaryOperation> T reduce(const BinaryOperation& binaryOperation, std::optional<T> initial = { }) const
```

Comment:
Please match the formatting used just above without the space after `template`. Ideally we would be consistent across all of WebKit but at least let’s stay locally consistent.

Can we find at least one place to use this besides the test? Normally we would not add something unused.

I’d prefer to have a more general purpose algorithm so it’s just as easy to use this on a span as on a vector, or on a `ListHasSet` or even a `HashSet`. So maybe it should be a non-member function. We don’t want to create more functions that lead people to unnecessarily create temporary vectors when what they want to operate on is already in some other kind of array or a subset of a vector or any other collection.

Reply:
Why add reduce() at all if there is no call site for it? I'd rather we introduce the reduce() function with some code calling it to make sure it is actually used and useful.